### PR TITLE
Adding question authority class for finding related works.

### LINKED ISSUE #1563: UI Create _find_work_widget partial
### Wireframes:
- [in context of New Work](https://wiki.duraspace.org/display/hydra/New+Work+-+tab+Relationships+-+Phase+1+-+UI+Wireframes#mockup_p1-work-new-tab_relationships) - Relationship Tab - sections This Work in Other Works, Other Works in This Work, Related Works
- [in context of Edit Work](https://wiki.duraspace.org/display/hydra/Edit+Work+-+Relationships+Tab+-+Phase+1+-+UI+Wireframes) - Relationship Tab - sections This Work in Other Works, Other Works in This Work, Related Works
- [in context of Batch Upload](https://wiki.duraspace.org/display/hydra/Batch+New+Works+-+tab+Relationships+-+Phase+1+-+UI+Wireframes) - Relationship Tab - sections These New Works in Other Works, Other Works in These New Works, Related Works
### Usage:

Used to find and select an existing work.  Options for finding a work:
- user begins typing the name of the work and auto-complete suggests works
- user chooses to search for works and can browse a list of all works with pagination
### Implementation:
- File: /app/views/generic_works/_find_work_widget.html.erb
- Create a partial for a view that includes the find_work_widget.
  - need to be able to name the field associated with the widget
  - need to be able to return a value for the field based on the user's selection
- Add class/methods that support auto-complete
- Add class/methods that support browse and select

### PR DESCRIPTION
**Pull request by andjsmit**
Fixes #1563 ; refs #1577

Use a local Question Authority to add autocomplete to Relationships: Other Works in this Work field

This PR is a long ways off of being ready. Added at this time to get help on a few points.
-  Can / Should a Search Builder be used in the local QA ?
-  How to get current work and user information into the QA ?

This work also has to be expanded to cover the This Work in Other Works field

@projecthydra/sufia-code-reviewers

### CODE REVIEW

File: app/views/curation_concerns/base/_find_work_widget.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<input class="new-form-control string multi_value optional related_works_ids work_child_members_ids form-control multi-text-field ui-autocomplete-input"
```
Comment (jcoyne) [inline app/views/curation_concerns/base/_find_work_widget.html.erb]: Is there a reason not to use the rails form helpers here? e.g. `f.input :ordered_member_ids, type: :text...`?

File: app/views/curation_concerns/base/_find_work_widget.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+       data-autocomplete-url="/authorities/search/local/works"
```
Comment (jcoyne) [inline app/views/curation_concerns/base/_find_work_widget.html.erb]: You should use the rails path helpers, so that it works if the application is not running at `/`

File: app/models/qa/authorities/local/find_works.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+module Qa::Authorities
+  class Local::FindWorks < ActiveFedora::Base
+    attr_reader :q
```
Comment (jcoyne) [inline app/models/qa/authorities/local/find_works.rb]: Is this ever used? Where?

File: app/assets/javascripts/sufia/autocomplete.es6 [hunk 4]
```diff
@@ -48,4 +51,9 @@ export class Autocomplete {
       var lang = require('sufia/autocomplete/language');
       new lang.Language(field, field.data('autocomplete-url'))
   }
+
+  autocompleteWork(field) {
+    var lang = require('sufia/autocomplete/work');
```
Comment (jcoyne) [inline app/assets/javascripts/sufia/autocomplete.es6]: can you pick a better name than `lang`?

File: app/assets/javascripts/sufia/autocomplete/work.es6 [hunk 5]
```diff
@@ -0,0 +1,24 @@
+export class Work {
```
Comment (jcoyne) [inline app/assets/javascripts/sufia/autocomplete/work.es6]: Add some class level documentation here.

File: app/views/sufia/admin_sets/_admin_set.html.erb [hunk 6]
```diff
@@ -21,5 +21,5 @@
   </td>
   <td><%= admin_set.create_date %></td>
   <td>N/A</td>
-  <td>N/A</td>
+  <td><%= controller.presenter_class.new(admin_set, current_ability).total_items %></td>
```
Comment (jcoyne) [inline app/views/sufia/admin_sets/_admin_set.html.erb]: This change seems unrelated to the rest of the PR, can you remove it?

File: spec/models/qa/authorities/local/find_works_spec.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+describe Qa::Authorities::Local::FindWorks do
+  let(:find_works) { FindWorks.new }
+  it "returns list of works" do
+    # TODO: Write test.
```
Comment (jcoyne) [inline spec/models/qa/authorities/local/find_works_spec.rb]: add something here.

File: app/views/sufia/admin_sets/_admin_set.html.erb (hunk 6 above)
Comment (andjsmit) [inline app/views/sufia/admin_sets/_admin_set.html.erb]: Ya, forgot that was in there. This line should have been included in a previous PR made. I'll pull it put and work it through a new issue.

File: spec/models/qa/authorities/local/find_works_spec.rb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+describe Qa::Authorities::Local::FindWorks do
+  subject { described_class.new(q: "Query") }
+
+  describe "#properties" do
+    it { is_expected.to respond_to("q") }
+    it { is_expected.to respond_to("search") }
```
Comment (mjgiarlo) [inline spec/models/qa/authorities/local/find_works_spec.rb]: Should this spec expand to test more of the `#search` method's behavior, or do you think this is covered well by other parts of the test suite?

**Comment by andjsmit** [comment]
Being replaced with new PR.
